### PR TITLE
dev(chore): log test names and suites

### PR DESCRIPTION
**Pull request by brad-decker**
## **Description**
Before #21441 the test reporter would output the test path it was executing which made figuring out where tests were failing at fairly easy. The updates in #21441 make tests run faster but it doesn't help when trying to ascertain where failures occur. To make it slightly easier this PR adds some reporting to the run-all script and the withFixtures and verboseReportOnFailure functions to explicitly call out where breakdowns happen.

This PR also changes from using `this.test.title` to `this.test.fullTitle()` which includes the name spacing of Describe Blocks for easier access. 

## **Manual testing steps**

1. View CircleCI output on develop for any of the E2E tests.
2. Observe that the test steps are written out verbosely, but no information about what file or test is being ran.
3. Do the same on this branch's CircleCI output and see the the test files and names are logged.
4. Repeat steps locally on develop and this branch to see local terminal console outputs file paths and test names. 

### **Before**
Screenshot 2023-10-26 at 1 48 27 PM

### **After**
Screenshot 2023-10-26 at 2 15 21 PM
Screenshot 2023-10-26 at 2 15 41 PM

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by HowardBraham** [comment]
Oh cool, I noticed that we lost that info when I changed the reporter.  Do you know if it's possible to go one better on the `Failure in ${title}, for more information see the artifacts tab in CI`, and actually get that error shown in the console?

**Comment by brad-decker** [comment]
@HowardBraham the details are already output, but they often don't make sense because its a stack trace. What is stored in test-artifacts are screenshots, a html document and the full output of the console in the browser which is better viewable in a file format

**Comment by HowardBraham** [comment]
Sorry, I wrote that in a confusing way.  Lemme give an example from the CircleCI run of this very PR: https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/61082/workflows/e50bdd66-7779-49bd-b1bc-16ba79536521/jobs/1912809/parallel-runs/2

The console log says
```
[driver] Called 'clickElement' with arguments ["[data-testid=\"snap-update-scroll\"]"]
Failure in can install an old and then updated version, for more information see the artifacts tab in CI
Error: Exited with code '1'
    at runInShell (/home/circleci/project/development/lib/run-command.js:134:29)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async /home/circleci/project/test/e2e/run-e2e-test.js:145:5
    at async retry (/home/circleci/project/development/lib/retry.js:38:22)
    at async main (/home/circleci/project/test/e2e/run-e2e-test.js:144:3)
Error: Retry limit reached
    at retry (/home/circleci/project/development/lib/retry.js:52:9)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async main (/home/circleci/project/test/e2e/run-e2e-test.js:144:3)
Error: Exited with code '1'
    at runInShell (/home/circleci/project/development/lib/run-command.js:134:29)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async main (/home/circleci/project/test/e2e/run-all.js:203:7)
```

If you click TESTS and expand the error you see
```
TimeoutError: Waiting for element to be located By(css selector, [data-testid="snap-update-scroll"])
Wait timed out after 10005ms
    at /home/circleci/project/node_modules/selenium-webdriver/lib/webdriver.js:929:17
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
```

If you're running locally, this info may or may not be in `test\test-results\e2e`, depending on how things go.  But it would be great if this TimeoutError were also on the console.

**Comment by brad-decker** [comment]
@HowardBraham ah, no that was me being dumb! thanks for pushing for better results, i think i got this done now i tested locally and get the error message in console.

File: test/e2e/webdriver/driver.js [hunk 1]
```diff
@@ -505,7 +505,17 @@ class Driver {
...
+    if (process.env.CIRCLECI) {
+      console.error(
+        `Failure in ${title}, for more information see the artifacts tab in CI\n`,
+      );
+    } else {
+      console.error(
+        `Failure in ${title}, for more information see the test-artifacts folder\n`,
+      );
+    }
+    console.error(`error: ${error.message}\n`);
```
Comment (HowardBraham) [inline test/e2e/webdriver/driver.js:518]: I can't decide if this line is better as
`console.error('error:', error);`

It then shows the stack trace, which would normally be better, but the line numbers are all wrong.

File: test/e2e/helpers.js
Comment (HowardBraham) [inline test/e2e/helpers.js:1]: I find that adding this on line 161 can be helpful
``console.log(`\nSuccess on test suite: ${title}\n`);``

File: test/e2e/webdriver/driver.js (hunk 1 above)
Comment (brad-decker) [reply test/e2e/webdriver/driver.js:518]: I still think the stack trace might be more beneficial, its still mroe information than just the message

File: test/e2e/helpers.js
Comment (brad-decker) [reply test/e2e/helpers.js:1]: Lemme know what you think of the change.

File: test/e2e/webdriver/driver.js (hunk 1 above)
Comment (HowardBraham) [reply test/e2e/webdriver/driver.js]: Do you have any inkling **why** the line numbers are wrong?  Minify or browserify?

**Review by HowardBraham — DISMISSED** [review]
Wow you changed 117 files and there's going to be tons of merge conflicts, but yeah the output is awesome now!

**Review by HowardBraham — APPROVED** [review]

**Review by darkwing — APPROVED** [review]

**Review by GuillaumeRx — APPROVED** [review]

**Pull request merged**